zookeeper/hparam_test.py

```diff
@@ -112,3 +112,20 @@ def test_str_nested(hyper_with_nested):
     )
 )"""
     assert click.unstyle(str(hyper_with_nested)) == output
+
+
+def test_init_kwargs(hyper):
+    new_hyper = hyper.__class__(foo="updated_foo", bar=-100, new_name="new_value")
```

Comment:
Is `.__class__` only necessary, because the fixture is already an instance?

Would `Hyper(foo="updated_foo", bar=-100, new_name="new_value")` or `HParams(foo="updated_foo", bar=-100, new_name="new_value")` work as well?

Reply:
Yeah, because it's already an instance.

I'd rather not use `HParams(...)` because I want to test that it correctly overrides values that already exist.

On the other hand using `Hyper(...)` would definitely be great, but I'd have to move the class definition outside of the method `hyper` to do so. Are you happy with that solution?

Reply:
> On the other hand using `Hyper(...)` would definitely be great, but I'd have to move the class definition outside of the method `hyper` to do so. Are you happy with that solution?

I think that's the cleanest solution. That way the intended usage is clear from looking at the unit test.